doc/algorithms.md

```diff
@@ -0,0 +1,727 @@
+# Core Algorithms used in MeetEval
+
+This file contains simplified implementations of the core algorithms used in the MeetEval toolkit.
+These implementations are meant for documentation.
+The actual implementations in the toolkit are more complex and track the alignment along with the distance computation, which is omitted here for clarity.
+
+All algorithms in this file operate on a _character level_ instead of the word level.
```

Comment:
Suggestion:

All algorithms in this file assume the input has the correct datatype, e.g., when the input is a string, the character errors are computed. To obtain the word errors, the input has to be a list of words.